Spanner/src/Database.php

```diff
@@ -210,32 +259,58 @@ class Database
      *        be returned as a {@see \Google\Cloud\Core\Int64} object for 32 bit
      *        platform compatibility. **Defaults to** false.
      * @param string $databaseRole The user created database role which creates the session.
+     * @param string $config [Optional] {
+     *     Configuration options.
+     *
+     *     @type bool $routeToLeader Enable/disable Leader Aware Routing.
+     *         **Defaults to** `true` (enabled).
+     *     @type array $defaultQueryOptions
+     * }
      */
     public function __construct(
-        ConnectionInterface $connection,
+        RequestHandler $requestHandler,
+        Serializer $serializer,
         Instance $instance,
-        LongRunningConnectionInterface $lroConnection,
         array $lroCallables,
-        $projectId,
-        $name,
+        string $projectId,
+        string $name,
         SessionPoolInterface $sessionPool = null,
-        $returnInt64AsObject = false,
+        bool $returnInt64AsObject = false,
         array $info = [],
-        $databaseRole = ''
+        string $databaseRole = '',
+        array $config = []
```

Comment:
I do not like extending our already bloated constructor with another parameter, especially one as generic as `$config`.

If anything , we should name this `$options` and combine it with all the other optional parameters.